janeczku/calibre-web
# Minimum viable implementation of the Kobo sync protocol

Introduce API endpoints for the Kobo sync protocol.

Most of the endpoints introduced are trivial placeholders for now, except for three endpoints:
- The v1/library/sync endpoint is used by the device to get a list of new/changed Books from the library since the last sync.
- The v1/library/<book_id>/metadata endpoint is used by the device to obtain a fresh download link for a given book.
- And a third trivial endpoint to get cover images.

This change introduces authentication changes that affects all of WebCalibre. It allows any user to login by passing a valid UserKey token instead of a user/password combination.

To test this change:
- The api_endpoint parameter on the device's .kobo/Kobo/Kobo eReader.conf file must be set to the CalibreWeb server address.
- The new "KoboStore UserKey" value under CalibreWeb's profile configuration should be set to the device's UserKey (found in the user table of .kobo/KoboReader.sqlite). I recommend reading up on the security caution notice in kobo_auth.py before doing so however.
- The new "config server url" value under Server configuration should be set to the server's address (e.g: 192.168.0.1:8083). (I've tried reconstructing this from the request url instead but it's fragile, and doesn't work when the server sits behind a proxy).

The Kobo device will now be able to sync (i.e catalog) all the book in the library, and download them locally for reading. Other features like Shelves, reading stats, etc are not yet implemented.
Known issues:
- This will also prevent the device from syncing to the official Kobo store.
